### ISSUE TITLE
Templates autocomplete for pom.xml files

### ISSUE THREAD

**Issue by akurtakov** [angelozerr/lsp4xml#517]
I'm using Eclipse WildWebDeveloper to edit pom.xml files (as it's generally better xml editor compared to m2e wtp based one - e.g. onhover help on closing tab and etc.) but I miss templates like https://git.eclipse.org/c/m2e/m2e-core.git/tree/org.eclipse.m2e.editor.xml/plugin.xml#n159 . Can lsp4xml provide them from the language server when editing pom.xml files?

**Comment by angelozerr** [comment]
@akurtakov it should be possible since LSP4XML provides extension to manage custom completion. But we must discuss with @fbricon about that. Because we should have too completion for dependencies for instance.

There are pro and cons to manage those snippet kind in LSP4XML:

 * pro: 
  * it will work with any client like vscode, lsp4e
  * we will able to to manage context to provide snippet only if there are inside a given element.
 * cons: not customizable like we have with Template preferences.

In other words we should discuss that with @fbricon

**Comment by akurtakov** [comment]
Was there any discussion on the topic?

**Comment by akurtakov** [comment]
@fbricon  was it discussed?

**Comment by angelozerr** [comment]
@akurtakov I'm afraid that it is not in our priority for the moment.

**Comment by AObuchow** [comment]
@angelozerr I'm very interested in trying out your work from the demo above. 
How did you integrate your maven extension for LSP4XML into WWD?
Did you declare in the package.json of vscode-xml the maven extension, and then use one of WWD's launch configurations to start the XML LS?

**Comment by AObuchow** [comment]
@angelozerr I was able to get your lsp4xml web extension working by following https://github.com/eclipse/wildwebdeveloper/issues/266#issuecomment-535386454, will try out your maven extension next week.

**Comment by angelozerr** [comment]
@AObuchow  please read https://github.com/angelozerr/lsp4xml-extensions-maven#on-wild-web-developer where I explain how you can launch the extension in LSP4E and where I explain how WWD could manage XML extension. If you have some questions about this topic, please create issues at https://github.com/angelozerr/lsp4xml-extensions-maven

Thanks!

**Comment by angelozerr** [comment]
I close this issue since https://github.com/angelozerr/lsp4xml-extensions-maven supports now maven template. I would like to move the SnippetRegistry class to LSP4XML once https://github.com/angelozerr/lsp4xml-extensions-maven will not more a POC

For maven support with LSP4XML, please create issues at https://github.com/angelozerr/lsp4xml-extensions-maven
